src/providers/wms/qgswmsprovider.cpp

```diff
@@ -482,7 +482,10 @@ bool QgsWmsProvider::setImageCrs( QString const &crs )
 void QgsWmsProvider::setQueryItem( QUrl &url, const QString &item, const QString &value )
 {
   url.removeQueryItem( item );
-  url.addQueryItem( item, value );
+  if ( value.isNull() )
+    url.addQueryItem( item, QStringLiteral( "" ) );
```

Comment:
QString()

Reply:
No: that is exactly the source of the problem: a null string will trigger the error.

Reply:
I don't know exactly when the problem appeared, the fact is that `QString()`.isNull() == true` and that's exactly the source of the issue.